apache/couchdb-fauxton
# Refactor documents/mango addon to use Redux

## Overview

Updated mango query to use Redux

Pending:
 - [x] Replace all uses of `typeOfIndex` with `docType`
 - [x] Introduce new `docType` for Mango query results
 - [x] Hide 'Metadata' button for Mango queries
 - [x] Remove `dispatch` from api.js files so they are not aware of Redux of any frameworks
 - [x] Cleanup unused code
 - [x] Resolve conflicts from master

## Testing recommendations

- Run Mango query
   - Switch between Table / JSON modes
   - Select & delete docs
   - Change 'Documents per page'
   - Navigate between pages using '<' and '>' buttons
- Click on 'manage index'
   - Index list should be displayed in the JSON mode
   - Create a new index. Check that index list is updated
   - Select & delete index
   - Change 'Documents per page'
   - Navigate between pages using '<' and '>' buttons

## Checklist
